packages/block-library/src/navigation-link/hooks.js

```diff
@@ -30,14 +30,14 @@ function getIcon( variationName ) {
 	}
 }
 
-function enhanceNavigationLinkVariations( settings, name ) {
+export function enhanceNavigationLinkVariations( settings, name ) {
 	if ( name !== 'core/navigation-link' ) {
 		return settings;
 	}
```

Comment:
cc @kjellr @shaunandrews if this branch doesn't work, it's much appreciated if folks add a `console.log( settings.variations )` here OR `debugger;` and send me a screenshot of what you see. Feel free to DM me if that didn't make sense.